outputs/jira.go

```diff
@@ -285,6 +272,113 @@ func createMetaProject(c *jira.Client, project string) (*jira.MetaProject, error
 	return metaProject, nil
 }
 
+func createIssueType(ctx *JiraAPI, metaProject *jira.MetaProject) error {
+	if ctx.Issuetype != "" {
+		for _, issueType := range metaProject.IssueTypes {
+			if ctx.Issuetype == issueType.Name { // search issueType in REST API
+				return nil
+			}
+		}
+		return fmt.Errorf("project %q don't have issueType %q", metaProject.Name, ctx.Issuetype)
+	} else {
+		for _, issueType := range metaProject.IssueTypes {
+			if issueType.Name == defaultIssueType { //search default issueType in REST API
+				ctx.Issuetype = defaultIssueType
+				return nil
+			}
+		}
+		if len(metaProject.IssueTypes) > 0 { // use 1st issueType from REST API
+			ctx.Issuetype = metaProject.IssueTypes[0].Name
+			return nil
+		} else {
+			return fmt.Errorf("project %q don't have issueTypes", metaProject.Name)
+		}
+	}
+}
+
+var createIssuePriority = func(ctx *JiraAPI, client *jira.Client) error {
+	issuePriorities, _, err := client.Priority.GetList()
+	if err != nil {
+		return fmt.Errorf("failed to get issue priority list: %w", err)
+	}
+	if ctx.Priority != "" {
+		for _, priority := range issuePriorities {
+			if ctx.Priority == priority.Name { // search priority in REST API
+				return nil
+			}
+		}
+		return fmt.Errorf("project don't have issue priority %q", ctx.Priority)
+	} else {
+		for _, priority := range issuePriorities {
+			if priority.Name == defaultIssuePriority { // search default priority in REST API
+				ctx.Priority = defaultIssuePriority
+				return nil
+			}
+		}
+		if len(issuePriorities) > 0 {
+			ctx.Priority = issuePriorities[0].Name // use 1st priority from REST API
+			return nil
+		} else {
+			return fmt.Errorf("project don't have issue priorities")
+		}
+	}
+}
+
+func createFieldsConfig(ctx *JiraAPI, client *jira.Client, content *map[string]string) (map[string]string, error) {
+	fields, _, err := client.Field.GetList()
+	if err != nil {
+		return nil, err
+	}
+
+	assignee := ctx.User
+	if len(ctx.Assignee) > 0 {
+		assignees := getHandledRecipients(ctx.Assignee, content, ctx.Name)
+		if len(assignees) > 0 {
+			assignee = assignees[0]
+		}
+	}
+
+	err = createIssuePriority(ctx, client)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create issue priority: %w", err)
+	}
+
+	fieldsConfig := make(map[string]string)
+
+	for _, field := range fields {
+		switch field.ID {
+		case "issuetype":
+			fieldsConfig[field.Name] = ctx.Issuetype
+		case "project":
+			fieldsConfig[field.Name] = ctx.ProjectKey
+		case "priority":
+			fieldsConfig[field.Name] = ctx.Priority
+		case "assignee":
+			fieldsConfig[field.Name] = assignee
+		case "description":
+			fieldsConfig[field.Name] = ctx.Description
+		case "summary":
+			fieldsConfig[field.Name] = ctx.Summary
+		}
+	}
+
+	// sprint is jira custom field.
+	// TODO make sprint field name check for non-English jira
+	if ctx.SprintId > 0 {
+		fieldsConfig["Sprint"] = strconv.Itoa(ctx.SprintId)
+	}
```

Comment:
Will `ctx.SprintId` be not set if it was non English jira?

Reply:
Sprint is a custom field. The Rest API returns the sprint ID as `customfield{Number}`.
We need more thoughtful logic for sprint.
I think we need to add it a bit later.

Reply:
I see - got it